### ISSUE TITLE
DAC_CHANNEL_MAX not declared

### ISSUE THREAD

**Issue by nitricware** [yellobyte/DacESP32#6]
Hy,

My very bare bones code gives me the following error: `/libraries/DacESP32/src/DacESP32.cpp:117:23: error: 'DAC_CHANNEL_MAX' was not declared in this scope; did you mean 'DAC_CHANNEL_2'?`

```
#include <DacESP32.h>

DacESP32 dac1(GPIO_NUM_25);

void setup() {
  // put your setup code here, to run once:
  Serial.begin(115200);
  Serial.println("Hello, ESP32!");
}

void loop() {
  // put your main code here, to run repeatedly:
  delay(10); // this speeds up the simulation
}
```

Thanks, Kurt

**Comment by yellobyte** [comment]
Hi Kurt,
when building for **ESP32** then DAC_CHANNEL_MAX is declared in "..../.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/hal/include/hal/dac_types.h" which should start like this:
```c
#pragma once

#include "soc/soc_caps.h"
#include "hal/adc_types.h"
#include "sdkconfig.h"

typedef enum {
    DAC_CHANNEL_1 = 0,    /*!< DAC channel 1 is GPIO25(ESP32) / GPIO17(ESP32S2) */
    DAC_CHANNEL_2 = 1,    /*!< DAC channel 2 is GPIO26(ESP32) / GPIO18(ESP32S2) */
    DAC_CHANNEL_MAX,
} dac_channel_t;
....
....
```
Building for **ESP32-S2**, **ESP32-S3** or **ESP32-C3** might **fail** for those MCUs **don't have a DAC module integrated**.
And of course, always make sure you use the latest ESP32 framework.
Cheers
Thomas

**Comment by nitricware** [comment]
I don't know what I'm missing.

I include the library via the Arduino IDE and upload the sketch to the ESP32. Other code ran fine that way.

I have the same problem with the Arduino IDE/physical ESP32 and Wokwi.

**Comment by yellobyte** [comment]
Make sure you have all recent updates for IDE, framework & libs. If no success then a clean IDE install might help.
Just did a build with settings as below.
![IDE Board Settings](https://github.com/user-attachments/assets/870f9dc6-0e61-4c00-ae7b-e32cf9f43326)
Here the complete build log just for reference:
[DacESP32-Build.log](https://github.com/user-attachments/files/16311900/DacESP32-Build.log)

**Comment by Jay-esp** [comment]
I have the same problem, the definition changed in the latest version of the esp32 libraries.

**Comment by yellobyte** [comment]
Ok, I'll have a look at it.

**Comment by yellobyte** [comment]
Just released **V1.1.1**. Please give it a try and have a look at the [release info](https://github.com/yellobyte/DacESP32/releases/tag/1.1.1).
